xtask/src/main.rs

```diff
@@ -7,41 +7,34 @@ mod zombienet;
 
 use clap::Parser;
 use cli::{test, Cli, Commands};
-use std::path::Path;
+use std::{
+    path::{Path, PathBuf},
+    str,
+};
 
 fn main() -> anyhow::Result<()> {
     init_logging()?;
     let cli = Cli::parse();
 
     match cli.command {
         Commands::Test(params) => test(params)?,
+        Commands::Zombienet(params) => zombienet(params)?,
     }
 
     Ok(())
 }
 
 fn test(params: test::Params) -> anyhow::Result<()> {
-    // extract project path
-    #[rustfmt::skip]
-    let project_path = duct::cmd!(
-        "sh", "-c",
-        "cargo metadata --format-version 1 | jq -r '.workspace_root'"
-    )
-    .stdout_capture()
-    .run()?;
-
-    let project_path = Path::new(std::str::from_utf8(&project_path.stdout)?.trim());
+    let project_path = obtain_project_path()?;
 
     init_env(&project_path, params.ci)?;
 
     build::build(&project_path, params.build)?;
 
     // the variables must be kept alive and not dropped
     // otherwise the child process will be killed
-    #[allow(unused)]
-    let zombienet = zombienet::Zombienet::try_new(&project_path, params.zombienet)?;
-    #[allow(unused)]
-    let shim = shim::Shim::try_new(&project_path, params.shim)?;
+    let _zombienet = zombienet::Zombienet::try_new(&project_path, params.zombienet)?;
```

Comment:
Wow, I did not know that `_variable` also prevents the variable from being dropped before the end of the scope! This is a lot better compared to the macro!